geocoding_android/android/src/main/java/com/baseflow/geocoding/Geocoding.java

```diff
@@ -34,26 +43,68 @@ void setLocaleIdentifier(@Nullable Locale locale) {
      * @return a list of Address objects. Returns null or empty list if no matches were found or there is no backend service available.
      * @throws java.io.IOException if the network is unavailable or any other I/O problem occurs.
      */
-    List<Address> placemarkFromAddress(String address) throws IOException {
-
+    @SuppressWarnings("deprecation")
+    void placemarkFromAddress(String address, GeocodeListenerAdapter callback) throws IOException {
         final Geocoder geocoder = createGeocoder(androidContext, locale);
-        return geocoder.getFromLocationName(address, 5);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            getAddressesWithGeocodeListener(geocoder, address, 5, callback);
+        } else {
+            List<Address> addresses = geocoder.getFromLocationName(address, 5);
```

Comment:
It would be nice to extract this line of code (which I believes causes the deprecation warning) into a separate method which we then decorate with the `@SuppressWarnings("deprecation")` attribute. This will ensure that the analysis will still pick up future deprecations within the rest of the method.

This would also simplify the error handling, since we will be able to handle the `IOException` locally (in this class) and can remove the redundant error handling (`try...catch` statements) in the `MethodCallHandlerImpl` class and rely on the `onError` callback method.

Something like:

```java
void placemarkFromAddress(String address, GeocodeListenerAdapter callback) {
    final Geocoder geocoder = createGeocoder(androidContext, locale);
        
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
        getAddressesWithGeocodeListener(geocoder, address, 5, callback);
    } else {
        try {
            List<Address> addresses = deprecatedGetFromLocationName(geocoder, address);
            callback.onGeocode(addresses);
        } catch (IOException ex) {
            callback.onError(ex.message);
        }
    }
}

@SuppressWarnings("deprecation")
private List<Address> deprecatedGetFromLocationName(Geocoder geocoder, String address) throws IOException {
    return geocoder.getFromLocationName(address, 5);
}
```